### DISCUSSION TITLE
Use custom components in VSCode

### DISCUSSION THREAD

**Discussion by cdia8890** [apache/camel-karavan#1323] [Q&A] [answered]
### Ask a question

Is there a possibility to add custom components to the Catalog UI by referring to a path in the local machine or any repository, rather than building the `components.json`?

The idea is to build camel components as spring boot starters and refer to those in other routes. 

Any advise for this is highly appreciated.

### Variant

Vs Code Extension

### Container Management (if applicable)

None

### Operating System (if applicable)

Windows

### Version

4.6.0

**Comment by davsclaus** [comment]
I dont think that this is possible in the current releases.

**Reply by cdia8890** [reply]
Is there a functionality proposed in the upcoming releases for this? Like to include an external jar which contains a camel route and view it in the UI?

Something like below and Karavan UI displays a component called `my-custom-starter`
`camel.jbang.depedencies=org.example:my-custom-starter:1.0.0`

I do understand the concept of Kamelets that they can be used but I'm more inclined to use a packaged file like a jar or something rather than a kamelet.yaml file which is hard to manage versioning.

The whole idea is to create reusable components as libraries and distribute across teams via a custom repository. So the teams can use those as components in their routes.

**Reply by davsclaus** [reply, marked as answer]
No there is no such upcoming feature.

**Reply by cdia8890** [reply]
ok thank you 👍
